catalog/admin/admin-poller-service/src/main/java/org/codice/ddf/catalog/admin/poller/AdminPollerServiceBean.java

```diff
@@ -215,6 +251,109 @@ public int compare(Map<String, Object> o1, Map<String, Object> o2) {
+                                                RegistryObjectMetacardType.REGISTRY_ID)
+                                                .equals(metacard.getAttribute(
+                                                        RegistryObjectMetacardType.REGISTRY_ID)))
+                                        .map(createdMetacard -> id)
+                                        .collect(Collectors.toList()));
+                            }
+                        } catch (IngestException e) {
+                            LOGGER.error("Unable to create metacard in catalogStore %s", id, e);
+                            Filter createFilter =
+                                    filterBuilder.attribute(RegistryObjectMetacardType.REGISTRY_ID)
+                                            .is()
+                                            .equalTo()
+                                            .text(source);
+                            Query createQuery = new QueryImpl(filter);
+                            SourceResponse createQueryResponse = catalogStoreMap.get(id)
+                                    .query(new QueryRequestImpl(createQuery));
+                            if (createQueryResponse.getHits() > 0) {
+                                //The metacard is still in the catalogStore, thus it was persisted
+                                updatedPublishedLocations.add(id);
+                            }
+                        }
+                    }
+                    for (String id : unpublishLocations) {
+                        DeleteRequest deleteRequest = new DeleteRequestImpl(metacard.getId());
+                        try {
+                            DeleteResponse deleteResponse = catalogStoreMap.get(id)
+                                    .delete(deleteRequest);
+                        } catch (IngestException e) {
+                            LOGGER.error("Unable to delete metacard from catalogStore %s", id, e);
+
+                            Filter deleteFilter =
+                                    filterBuilder.attribute(RegistryObjectMetacardType.REGISTRY_ID)
+                                            .is()
+                                            .equalTo()
+                                            .text(source);
+                            Query deleteQuery = new QueryImpl(filter);
+                            SourceResponse deleteQueryResponse = catalogStoreMap.get(id)
+                                    .query(new QueryRequestImpl(deleteQuery));
+                            if (deleteQueryResponse.getHits() > 0) {
+                                //The metacard is still in the catalogStore, thus wasn't unpublished
+                                updatedPublishedLocations.add(id);
+                            }
+
+                        }
+                    }
+
+                    metacard.setAttribute(new AttributeImpl(RegistryObjectMetacardType.PUBLISHED_LOCATIONS,
```

Comment:
If you implement my suggestion above then  you will need to add the union of destinations and currentPublishedLocations to the updatedPublishedLocations